### DISCUSSION TITLE
I am trying to port my electron app to Tauri. I am creating a POC First. Can anyone tell me how feasible it is

### DISCUSSION THREAD

**Discussion by DVGY** [tauri-apps/tauri#7096] [Q&A] [answered]
These are the question I want to ask Tauri Community and Great developer behind it.

Trying create an offline app, which store data on disk and later can sync to some cloud database.

1. Is it possible to store images on disk (1 to 6 MB in size) efficiently ?
2. Is it possible to store files (pdf, xlsx, docx) on disk (1 to 10 MB is size) efficiently ?
3. Store data in some sql like sql-lite or pg-sql.
4. Images resizing.
5. Multiple Windows and communications btw them.

**Comment by FabianLars** [comment, marked as answer]
Reposting my response from [discord](https://discord.com/channels/616186924390023171/1113397060083920997):
```
yes, but you probably won't get around using Rust, in case that's a problem.

To be a bit more throughout: 

> Is it possible to store images on disk (1 to 6 MB in size) efficiently ?
> Is it possible to store files (pdf, xlsx, docx) on disk (1 to 10 MB is size) efficiently ?

Depends on what you mean with efficiently. The file size completely depends on your code (are you compressing it etc). The only issue is that v1's IPC is pretty slow so if you plan to get that data from the js side into the file system than it _may_ be better to wait for v2 which will have a faster IPC.

> Store data in some sql like sql-lite or pg-sql.

we have a sqlite plugin for js users, but for more complicated uses you'll need to use rust.

> Images resizing.

That got nothing to do with Tauri, there are libs in both js and rust that will help you with that.

> Multiple Windows and communications btw them.

Won't be a problem as long as all windows show local pages (no external urls)
```

**Reply by DVGY** [reply]
Okay. 
It is possible to write a rust code for us. 

Do you think we can probably mix rust and js code 😄. For example I have a page, in which user enters some text field, upload an image from his computer and few files. Can I write a rust code for this page specifically inside react/js file ? or need to depend on IPC or calling rust func?

In short is it possible to use rust entirely for some pages? and it should be usable in react/js ? 

Thanks

**Reply by FabianLars** [reply]
> Can I write a rust code for this page specifically inside react/js file ?

No

> or need to depend on IPC or calling rust func?

Yes

Think of the rust side/backend as something similar to a remote rest api (at least that's how i always think of it). Also it's not about the language itself here, otherwise you could use Rust and compile it to WebAssembly. The issue here is that javascript/webassembly run in the browser's sandbox so they do not have direct access to the operating system and it's file system etc.

**Reply by DVGY** [reply]
Ohh that's a good one and it never came to my mind. After looking at the documentation it make sense 
![image](https://github.com/tauri-apps/tauri/assets/46158826/73ea3c17-69c6-4f96-b37c-b672b98875d1)
only these tech stack can be used as front end. 

Okay thanks alot. Really appreciate it

**Reply by FabianLars** [reply]
> only these tech stack can be used as front end.

Well, no. You can use pretty much everything you can use for traditional browser-targeted frontends as long as they don't need an actual server for Server-Side Rendering.
We just don't have a guide for every single frontend framework out there which is what `Integrate into Existing Project` tries to make up for.

The Rust "backend" is basically the replacement of a traditional webserver/nodejs runtime or electron's `main` process.

**Reply by simonhyll** [reply]
I wonder how we can make it clearer that those aren't the only supported frameworks, lots of people seem to come to that conclusion

Any framework that supports static hosting can be used with Tauri. Imagine your project had to be launched on an Apache or Nginx server. If the framework you're using supports that, it's supported by Tauri.

Shameless self promotion, here's a video about what frontend agnostic means: https://youtu.be/C5k05-YUrqU

**Reply by DVGY** [reply]
Nice I just edited my POC and put your video link.

**Reply by DVGY** [reply]
@simonhyll  Maybe adding a section for other framework in the documentation and then explain all this agnostic thing
